crates/turbo-tasks-fetch/src/lib.rs

```diff
@@ -29,7 +32,11 @@ impl HttpResponseBodyVc {
 }
 
 #[turbo_tasks::function]
-pub async fn fetch(url: StringVc, user_agent: OptionStringVc) -> Result<HttpResponseVc> {
+pub async fn fetch(
+    url: StringVc,
+    user_agent: OptionStringVc,
+    context: FileSystemPathVc,
```

Comment:
It seems weird that we would pass in the context just for the error case. Could `fetch` return a `Result<FetchResultVc>` which is a `Result<HttpResponseVc, FetchIssueVc>`. The caller could then manually report the issue with:

```rust
let response = match fetch_result.await? {
    Ok(response) => response,
    Err(issue) => {
        issue.as_issue().with_context(context).emit();
        bail!(...);
    }
};
```

I wish we could generalize this pattern, but without generic support in `Vc`s, I don't see how that's tractable...